libs/collab-stream/src/stream_router.rs

```diff
@@ -198,21 +213,27 @@ impl Worker {
     keys: &mut Vec<StreamKey>,
     ids: &mut Vec<String>,
     senders: &mut HashMap<&str, (StreamSender, usize)>,
-  ) {
+  ) -> usize {
     let keys = keys.drain(..);
     let mut ids = ids.drain(..);
+    let mut scheduled = 0;
     for key in keys {
       if let Some(last_id) = ids.next() {
         if let Some((sender, _)) = senders.remove(key.as_str()) {
+          if sender.is_closed() {
```

Comment:
The problem was when there were no new messages in Redis stream. In such case we didn't check if any active subscriber was actually requesting for them. If subscriber was closed we still tried to read the stream even if nobody wanted to listen to it.

This could mean that if we had 5000 subscribers in the past, but no new data was coming from their Redis streams, and only 5 of them where active atm. we still tried to XREAD 5000 streams instead of 5.